devtempstats_service/native/src/impl/DevTempStatsService.cpp

```diff
@@ -0,0 +1,272 @@
+      {
+
+         *_aidl_return = cpu_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getGpuTemperature(float *_aidl_return)
+      {
+
+         *_aidl_return = gpu_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getAmbientTemperature(float *_aidl_return)
+      {
+
+         *_aidl_return = ambient_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getAverageCpuTemperature(float *_aidl_return)
+      {
+
+         *_aidl_return = average_cpu_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getAverageGpuTemperature(float *_aidl_return)
+      {
+
+         *_aidl_return = average_gpu_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getAverageAmbientTemperature(float *_aidl_return)
+      {
+
+         *_aidl_return = average_ambient_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getMaxCpuTemperature(float *_aidl_return)
+      {
+
+         *_aidl_return = max_cpu_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getMaxGpuTemperature(float *_aidl_return)
+      {
+         *_aidl_return = max_gpu_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      ::ndk::ScopedAStatus DevTempStatsService::getMaxAmbientTemperature(float *_aidl_return)
+      {
+         *_aidl_return = max_ambient_value();
+         return ndk::ScopedAStatus::ok();
+      }
+
+      void DevTempStatsService::ReadColumnFromCSV(const std::string &filename, int col_index, std::vector<float> &column_data)
+      {
+         std::ifstream file(filename);
+         std::string line;
+
+         if (file.is_open())
+         {
+            while (getline(file, line))
+            {
+               std::stringstream ss(line);
+               std::string cell;
+               int current_col = 0;
+               while (getline(ss, cell, ',') && current_col <= col_index)
+               {
+                  if (current_col == col_index)
+                  {
+                     column_data.push_back(std::stof(cell));
+                     break; // Stop reading after finding the desired column
+                  }
+                  ++current_col;
+               }
+            }
+            file.close();
+         }
+         else
+         {
+            std::cerr << "Unable to open file: " << filename << std::endl;
+         }
+      }
+
+      float DevTempStatsService::GetCpuNextValue(std::vector<float> &vec)
+      {
+         static std::size_t index_cpu = 0;
+
+         if (index_cpu >= vec.size())
+         {
+            index_cpu = 0;
+         }
+
+         return vec[index_cpu++];
```

Comment:
versiunea `1337` pentru a elimina if-ul este 
```suggestion
         return vec[++index_cpu % vec.size()];
```
Nu este tot timpul recomandata folosirea acestui tip de cod(combinarea de instructiuni pentru a reduce numarul de linii de cod, altfel numit leet coding), atunci cand reduce din "code readability"